lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -731,15 +734,28 @@ def do_3d_projection(self):
         if len(self._linewidths3d) > 1:
             self._linewidths = self._linewidths3d[z_markers_idx]
 
+        PathCollection.set_offsets(self, np.column_stack((vxs, vys)))
+
         # Re-order items
         vzs = vzs[z_markers_idx]
         vxs = vxs[z_markers_idx]
         vys = vys[z_markers_idx]
 
-        PathCollection.set_offsets(self, np.column_stack((vxs, vys)))
+        # Store ordered offset for drawing purpose
+        self._offset_zordered = np.column_stack((vxs, vys))
```

Comment:
Can you make sure that this gets created in `set_3d_properties` as well so that we can be sure to never end up in a situation where the context manager will fail with `AttributeError`?

Reply:
> Can you make sure that this gets created in `set_3d_properties` as well so that we can be sure to never end up in a situation where the context manager will fail with `AttributeError`?

I'd rather check the attribute existance by `hasattr` or an `None` check with an initialization with `None` initialize self._offset_zordered with None in initialization to secure the `AttributeError` problem. But, I don't think `set_3d_properties` ever set offset projection. Do you mean we want to execute `do_3d_projection` in the end of `set_3d_properties`?

Reply:
Checking for `None` is better than `hasattr` (just because we _can_ dynamically add attributes does not mean we should).